### PR TITLE
pkg/k8s: stop logging conflicting errors as errors on update

### PR DESCRIPTION
**Pull request by aanm**
Signed-off-by: André Martins <andre@cilium.io>

Fixes: #4674

```release-note
stop logging conflicting errors as errors when modifying kubernetes objects
```

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/4676)

### CODE REVIEW

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

File: pkg/k8s/client.go [hunk 1]
```diff
@@ -189,21 +189,26 @@ func AnnotateNode(c kubernetes.Interface, nodeName string, v4CIDR, v6CIDR *net.I
...
+			case errors.IsConflict(err):
+				scopedLog.WithFields(logrus.Fields{
+					fieldRetry:    n,
+					fieldMaxRetry: maxUpdateRetries,
+				}).WithError(err).Debugf("Unable to update node resource with annotation")
+			default:
 				scopedLog.WithFields(logrus.Fields{
 					fieldRetry:    n,
 					fieldMaxRetry: maxUpdateRetries,
 				}).WithError(err).Error("Unable to update node resource with annotation")
```
Comment (tgraf) [inline pkg/k8s/client.go]: Let's make this a warning as we retry right away

**Review by tgraf — CHANGES_REQUESTED** [review]

File: pkg/k8s/client.go (hunk 1 above)
Comment (aanm) [reply pkg/k8s/client.go]: Done

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-K8s/445/ DNS known flake from cilium to master. re-triggering it again after a rebase from master which containers microscope to be setup in the this test.

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Review by tgraf — APPROVED** [review]

**Pull request merged**